### PR TITLE
feat: share slackbot

### PR DESCRIPTION
**Pull request by rephus**
Closes: https://github.com/lightdash/lightdash/issues/2776
### Description:

Typescript implementation https://github.com/slackapi/bolt-js/issues/904#issuecomment-1032175818

New slack docs https://deploy-preview-3462--peaceful-bassi-cbf284.netlify.app/guides/sharing-in-slack

> Because this integration is useless until we have headless-chrome, I used a localstorage variable `localStorage.getItem('slack')` to hide the slack settings . Everything else can be merged.

[Screencast from 21-11-22 16:41:12.webm](https://user-images.githubusercontent.com/1983672/203277568-f6529608-c255-4500-a586-31a80b48e57f.webm)

Screenshot from 2022-11-22 10-06-18
Screenshot from 2022-11-22 10-19-08
Screenshot from 2022-11-22 10-22-54

### CODE REVIEW

**Comment by rephus** [comment]
Fixing migration on render

**Comment by rephus** [comment]
Tested on this render, I had to manually add redirect on slack settings 

Screenshot from 2022-11-22 12-07-44

**Comment by rephus** [comment]
we can merge this into master, then once headless-chrome is done, we can replace the `unfurl` placeholder with the proper dashboard screenshot. There are also a few extra things that can be done on a separate PR before merging. 

https://github.com/lightdash/lightdash/issues/3821

File: packages/backend/src/clients/Slack/Slackbot.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                        accessory: {
+                            type: 'button',
+                            text: {
+                                type: 'plain_text',
+                                text: 'Open dashboard',
+                                emoji: true,
+                            },
+                            value: 'view_alternate_1',
+                        },
+                    }, */
```
Comment (IrakliJani) [inline packages/backend/src/clients/Slack/Slackbot.ts:103]: this chunk will be enabled after we deploy headless chrome right? before that we can just test that integration is working?

File: packages/frontend/src/components/UserSettings/SlackSettingsPanel/SlackSettingsPanel.styles.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export const ColorPalette = styled.div`
+    display: grid;
+    grid-template-columns: 1fr 1fr 1fr;
+    column-gap: 40px;
+`;
+
+export const ColorSquare = styled.div`
+    height: 30px;
+    width: 30px;
+    padding: 0.286em;
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/SlackSettingsPanel/SlackSettingsPanel.styles.ts:22]: I find it really hard to maintain having em's in the codebase. any benefit of using it?

File: packages/frontend/src/components/UserSettings/SlackSettingsPanel/index.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                    href={`/api/v1/slack/install/${user.data?.organizationUuid}`}
+                    target="_blank"
+                    rel="noreferrer"
+                >
+                    <img
+                        alt="Add to Slack"
+                        height="40"
+                        width="139"
+                        src="https://platform.slack-edge.com/img/add_to_slack.png"
+                        srcSet="https://platform.slack-edge.com/img/add_to_slack.png 1x, https://platform.slack-edge.com/img/add_to_slack@2x.png 2x"
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/SlackSettingsPanel/index.tsx:61]: can we have a regular blueprint button with slack icon in it?

**Review by IrakliJani — COMMENTED** [review]
nice job, backend code looks great 🎉 
added a few comments in the frontend side

File: packages/backend/src/clients/Slack/Slackbot.ts (hunk 1 above)
Comment (rephus) [reply packages/backend/src/clients/Slack/Slackbot.ts:103]: yes, right now there are no images to display. we might also add a few extra blocks

File: packages/frontend/src/components/UserSettings/SlackSettingsPanel/SlackSettingsPanel.styles.ts (hunk 2 above)
Comment (rephus) [reply packages/frontend/src/components/UserSettings/SlackSettingsPanel/SlackSettingsPanel.styles.ts:22]: Do you mean `em`s ? These tags are not even used, it was a copypaste leftover. I'll remove them.

File: packages/frontend/src/components/UserSettings/SlackSettingsPanel/SlackSettingsPanel.styles.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+`;
+
+export const ColorSquareInner = styled.div`
+    height: 100%;
+    width: 100%;
+    border-radius: 1px;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+`;
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/SlackSettingsPanel/SlackSettingsPanel.styles.ts:32]: I feel like most of the css from this file is not relevant to this page and can be removed
CleanShot 2022-11-23 at 14 33 10@2x

File: packages/backend/src/clients/Slack/Slackbot.ts (hunk 1 above)
Comment (IrakliJani) [reply packages/backend/src/clients/Slack/Slackbot.ts:103]: nice!

**Pull request merged**